ion/textwriter.go

```diff
@@ -336,9 +407,15 @@ func (w *textWriter) begin(api string, t ctx, c byte) error {
 	}
 
 	w.ctx.push(t)
+	w.indent++
 	w.needsSeparator = false
+	w.emptyContainer = true
 
-	return writeRawChar(c, w.out)
+	if err := writeRawChar(c, w.out); err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
A quick Google search did not give me an answer here. Can this block of code stay the same, i.e., 

```
return writeRawChar(c, w.out)
```

Is there a reason for checking for the error and returning, as opposed to delegating to the `writeRawChar` function that I believe also returns `nil` or `err`?

Reply:
Ah, yeah, we can just return. I had added some code after the call to `writeRawChar` at some intermediate point, then backed it out but didn't simplify. You'll sometimes see needlessly-explicit error checking like this instead of a straight return so there's less refactoring back and forth as the code changes; I _think_ this part of the code should be relatively stable now though.